### ISSUE TITLE
Investigate using a mechanism other than `whoami` to determine username

### ISSUE THREAD

**Issue by dustymabe** [dustymabe/vagrant-sshfs#1]
We investigate using a mechanism other than `whoami` to determine username. We do that [here](https://github.com/dustymabe/vagrant-sshfs/blob/2576a0243bc08a6f2fc1a818225ae9ef281ef375/lib/vagrant-sshfs/synced_folder.rb#L116)

**Comment by codylane** [comment]
@dustymabe - I'm curious if you've tried using `ENV['USER']` instead of `whoami` for this?  I'd be happy to give this a shot and see if this still works as well as put some tests behind it?

**Comment by codylane** [comment]
There is also the `etc` ruby module that should be a much richer alternative as well.  I'd suggest using the `etc` module route and here's an example.


```
[1] pry(main)> require 'etc'
=> false

[2] pry(main)> Etc.getlogin
=> "codylane"

[3] pry(main)> Etc.getpwnam(Etc.getlogin)
=> #<struct Etc::Passwd name="codylane", passwd="********", uid=501, gid=20, gecos="codylane", dir="/Users/codylane", shell="/bin/bash", change=0, uclass="", expire=0>

[4] pry(main)> Etc.getpwnam(Etc.getlogin).uid
=> 501

[5] pry(main)> Etc.getpwnam(Etc.getlogin).gid
=> 20
```

Here's the [etc API doc ](https://docs.ruby-lang.org/en/2.6.0/Etc.html)for reference for 2.6.x which is the version of ruby what Vagrant 2.2.x is using

**Comment by dustymabe** [comment]
Seems reasonable assuming that module is in the standard library for ruby (i.e. it's not something extra that needs to be installed later).

**Comment by codylane** [comment]
Agreed, for confirmation [etc](https://ruby-doc.org/stdlib-2.6.6/) is part of the standard library. I'll put in a PR with some tests when before the end of the week.
